### PR TITLE
Add workflow for releasing vagrant_cloud gem

### PR DESCRIPTION
**Pull request by briancain**

### CODE REVIEW

File: .github/workflows/gempush.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      run: |
+        mkdir -p $HOME/.gem
+        touch $HOME/.gem/credentials
+        chmod 0600 $HOME/.gem/credentials
+        printf -- "---\n:github: Bearer ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+        gem build *.gemspec
+        gem push --KEY github --host https://rubygems.pkg.github.com/${OWNER} *.gem
+      env:
+        GEM_HOST_API_KEY: ${{secrets.GPR_AUTH_TOKEN}}
+        OWNER: username
```
Comment (jbonhag) [inline .github/workflows/gempush.yml:29]: I can't seem to find documentation for this `username` value -- does it exist and I'm just not seeing it? Or do we need to pull the username out of [GITHUB_REPOSITORY](https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables)?

Comment (briancain) [reply .github/workflows/gempush.yml:29]: Oops. I think this probably just needs to be updated to the proper username? I think it's `hashicorp-vagrant`

File: .ci/common.sh [hunk 2]
```diff
@@ -375,6 +365,85 @@ function hashicorp_release() {
...
+
+    # Install dependency if required
+    gem list reaper-man --installed > "${output}"
+    if [ $? -ne 0 ]; then
+        wrap_stream gem install reaper-man --no-document \
+                    "Failed to install dependency for hashigem generation"
+    fi
+
+    # Create a temporary directory to work from
+    tmpdir="$(mktemp -p ./)" ||
```
Comment (jbonhag) [inline .ci/common.sh:404]: I believe this needs a `-d` flag.

Comment (chrisroberts) [reply .ci/common.sh:404]: Updating to the latest version includes this fix

**Pull request merged**